log/logfmt_logger.go

```diff
@@ -18,16 +38,45 @@ func NewLogfmtLogger(w io.Writer) Logger {
 }
 
 func (l logfmtLogger) Log(keyvals ...interface{}) error {
-	// The Logger interface requires implementations to be safe for concurrent
-	// use by multiple goroutines. For this implementation that means making
-	// only one call to l.w.Write() for each call to Log. We first collect all
-	// of the bytes into b, and then call l.w.Write(b).
-	b, err := logfmt.MarshalKeyvals(keyvals...)
-	if err != nil {
+	// Code copied from logfmt.MarshalKeyvals and modified
+	// to reuse encoders and buffers with sync.Pool and
+	// call EndRecord
```

Comment:
Yes, although I think adding a method to `logfmt.Encoder` makes more sense.

```go
func (enc *Encoder) EncodeKeyvals(keyvals ...interface{}) error
```

See: https://github.com/go-logfmt/logfmt/commit/a0ff333161fe5c2daed0ba52d1792bd3d2531b94.